### ISSUE TITLE
Xử lý dữ liệu

### ISSUE THREAD

**Issue by manhcuong17072002** [nguyenthienhy/F5-TTS-Vietnamese#5]
Cho mình hỏi, trước khi fine tune model thì bạn có xử lý dữ liệu vivoice hay vlsp không ạ. Và bạn xử lý như nào vậy ạ. Cảm ơn

**Comment by nguyenthienhy** [comment]
Ngoài các kỹ thuật xử lý đã nói ở repos model huggingface:
Remove all music background from audios, using facebook demucs model: https://github.com/facebookresearch/demucs
Do not use audio files shorter than 1 second or longer than 30 seconds.
Keep the default punctuation marks unchanged.
Normalize to lowercase format.
Mình có một vài kỹ thuật xử lý sâu nữa nhé, sắp tới mình sẽ đăng một bài nói riêng về phần này. Thân gửi tới bạn.

**Comment by manhcuong17072002** [comment]
Mình đang finetune và có loss như sau:

![Image](https://github.com/user-attachments/assets/81fdd313-b03a-4e3e-a763-c35d18a3c92d)

Cấu hình mình đang sử dụng: 
- batchsize: 12800 frame
- warmup: 500
- lr: 1e-5
- epoch: 10

Bạn có thể cho chút nhận xét so với quá trình bạn fine tune được không ạ.

**Comment by nguyenthienhy** [comment]
Bạn train bao nhiêu giờ dữ liệu nhỉ ?

**Comment by manhcuong17072002** [comment]
@nguyenthienhy Hiện tại mình train bộ vivoice sau khi lọc đoạn ngắn dài thì khoảng > 800h ạ. Và mình không làm bước này: 
- Remove all music background from audios, using facebook demucs model: https://github.com/facebookresearch/demucs

Nhưng mình thêm bước: 
- Chuẩn hóa số thành text

**Comment by nguyenthienhy** [comment]
warmup nên để = 3 đến 10% số steps
nếu train với bộ vivoice thì loss như trên là hoàn toàn bình thường nhé, do transcription dùng whisper-large-v3 gen nên phân bố data sẽ có một lượng nhiễu nhất định

**Comment by manhcuong17072002** [comment]
@nguyenthienhy Cho mình hỏi làm sao để biết ckpts nó đã tốt hay chưa trong khi sample_audio_gen thi thoảng nó vẫn có một số lỗi đọc sai ạ. Với cả mình thấy loss nó bị dao động nhiều quá thì có hợp lý không nhỉ. Mình đã fine tune khoảng 10 epochs rồi nhưng kết quả vẫn khá tệ ạ.

**Comment by nguyenthienhy** [comment]
1. khá tệ là thế nào bro nhỉ ?
2. một số lỗi đọc sai thì nó phụ thuộc vào audio ref của bạn nữa. một tips là nên chọn audio ref nhỏ hơn 10s, đọc rõ ràng.
3. phần text cần gen thì nên chuẩn hoá dấu câu, text norm thêm nhé

**Comment by manhcuong17072002** [comment]
@nguyenthienhy Mình có một số phản hồi như sau:
1. Khá tệ như: Đọc sai chữ, đọc x2 chữ nào đó, hoặc đọc không rõ. Còn chất lượng giọng clone thì ổn ạ
2. Audio ref hiện tại của mình rất tốt ạ. Mình có nhờ MC với TW để thu âm hộ. Thấy giọng clone khá tốt và đều nhau
3. Mình có dùng vinorm chuẩn hóa cơ bản trước khi đưa vào (vì mình thử các mẫu đơn giản) rồi. Nhưng kết quả không khả quan lắm ạ 

Cảm ơn ạ

**Comment by nguyenthienhy** [comment]
1. ở đây mình đoán bro infer một đoạn văn khá dài thì phải, bro thử tách câu ra infer xem hiện trạng
2. hiện tượng đọc sai chữ, x2 chữ, đọc không rõ là có, nhưng không phải giọng nào cũng bị như vậy => bro nên kiếm thêm một số tập dataset có chất lượng transcription thật tốt, training cùng bộ vivoice

**Comment by manhcuong17072002** [comment]
Để mình thử thêm xem sao ạ
